elasticsearch/init/0100-seed-index-templates

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash -e
+#
+# Copyright 2018 Red Hat, Inc. and/or its affiliates
+# and other contributors as indicated by the @author tags.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source "logging"
+
+info Adding index templates
+shopt -s failglob
+for template_file in ${ES_HOME}/index_templates/*.json
+do
+    sed -i "s,\$REPLICA_SHARDS,$REPLICA_SHARDS," $template_file
+    sed -i "s,\$PRIMARY_SHARDS,$PRIMARY_SHARDS," $template_file
+    template=`basename $template_file`
+    # Check if index template already exists
+    response_code=$(es_util --query=_template/$template \
+        ${DEBUG:+-v} -s \
+        --request HEAD --head --output /dev/null \
+        -w '%{response_code}')
+    if [ "${response_code}" == "200" ]; then
+        info "Index template '$template' found in the cluster, overriding it"
+    else
+        info "Create index template '$template'"
+    fi
+    es_util --query=_template/$template \
+        curl ${DEBUG:+-v} -s -X PUT \
```

Comment:
is `$ES_REST_BASEURL` missing here?

Reply:
yes i believe it is.  This assumes you are running against local host which will not be the case of run as an init container

Reply:
We will need to pass `$ES_BASE` as an env var to the init container which should be "https://logging-es:9200" for template initialization and 
$ES_HOST AND $ES_PORT for the seeding to run. $ES_HOST is `logging-es-cluster` service and port defaults to `9300`.  Maybe we can source these from env vars

Reply:
We should pass the values of:
```
ES_HOST=${LOGGING_ES_CLUSTER_SERVICE_HOST}
ES_PORT=${LOGGING_ES_CLUSTER_SERVICE_PORT}
```
I refrain from adding those here until we move init out of each ES node since then it would force it to go through the service instead of `localhost`.  The vars passed by the kublet are more appropriately named.

Reply:
Further looking we will also need to base $ES_BASE so the es_util script will succeed which can also be changed for completeness to:
```
ES_BASE="${LOGGING_ES_SERVICE_HOST}:${LOGGING_ES_SERVICE_PORT}"
```